torch/csrc/jit/script/module.cpp

```diff
@@ -113,11 +113,10 @@ void Module::to_impl(
   for (auto& parameter : get_parameters()) {
     // Need to access the `at::Tensor` as a `Variable` here.
     autograd::Variable variable = parameter.value().toTensor();
-    at::Tensor data = variable.data();
-    // Use the data's original device or dtype if not supplied here.
-    auto new_data = data.to(
-        device.value_or(data.device()),
-        dtype.value_or(data.scalar_type()),
+    // Use the variable's original device or dtype if not supplied here.
+    auto new_data = variable.to(
+        device.value_or(variable.device()),
+        dtype.value_or(variable.scalar_type()),
```

Comment:
It's okay to use `variable` instead of `variable.data()` here, because `new_data` is being used in the `set_data(...)` call below, and `set_data(...)` ignores `new_data`'s AutogradMeta.